File: app/src/main/java/ru/agpu/artikproject/background_work/service/BootBroadcast.java
package ru.agpu.artikproject.background_work.service;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.os.Build;
import android.util.Log;

public class BootBroadcast extends BroadcastReceiver {

    @Override
    public void onReceive(Context context, Intent intent) {
        Log.i("onReceive", "Обн.: " + intent.getAction());
        try {
            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
                context.startForegroundService(new Intent(context, PlayService.class));
            } else {
                context.startService(new Intent(context, PlayService.class));
            }
        } catch (Exception e) {
            Log.e("BootBroadcast", "onReceive error");
            e.printStackTrace();
        }
    }
}

File: app/src/main/java/ru/agpu/artikproject/background_work/service/BootBroadcast.kt
package ru.agpu.artikproject.background_work.service

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.os.Build
import android.util.Log

class BootBroadcast: BroadcastReceiver() {

    override fun onReceive(context: Context, intent: Intent) {
        Log.i("onReceive", "Обн.: " + intent.action)
        try {
            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
                context.startForegroundService(Intent(context, PlayService::class.java))
            } else {
                context.startService(Intent(context, PlayService::class.java))
            }
        } catch (e: Exception) {
            Log.e("BootBroadcast", "onReceive error")
            e.printStackTrace()
        }
    }
}